### PR TITLE
[FEATURE] Stable build V1 (ChromeExtension + FastAPI Server + Fine tuned BART model integration)

### PR DESCRIPTION
**Pull request by rootsec1**
Implemented Email Summarization using BART Model

## Description:
This pull request implements an email summarization system leveraging the power of BART (Bidirectional and Auto-Regressive Transformers) model from Facebook. The system fetches emails from the user's inbox over the past week, extracts relevant content, generates a concise summary using BART, and sends it back to the user via email.

## Changes Made:
1. **Implementation of EmailTLDR Class**: Introduced the `EmailTLDR` class, facilitating email retrieval, content extraction, summarization, and email sending functionalities.

2. **Email Retrieval and Extraction**: Implemented methods to connect to the user's email server securely, fetch emails from the past week, and extract relevant content including email bodies.

3. **BART Model Integration**: Utilized the `facebook/bart-base` model for email summarization. The model is fine-tuned to generate succinct summaries of email content.

4. **Summarization and Email Sending**: Developed the `generate_summary` method to process email content and generate summaries using the fine-tuned BART model. Integrated functionality to compose and send summary emails to the user's email address.

#### Usage:
- To use the email summarization system, instantiate the `EmailTLDR` class with your email credentials.
- Call the `summarize_and_send` method to fetch, summarize, and send email summaries from the past week.

## Example Usage:
```python
# Instantiate EmailTLDR object
email_tldr = EmailTLDR(email_address, email_password)

# Summarize and send emails from the past week
email_tldr.summarize_and_send()
```

## Additional Notes:
- Ensure proper configuration of environment variables for email credentials.
- The system can be further extended to incorporate advanced summarization techniques or handle specific email formats.
- Suggestions and feedback for improvement are welcome.

## Testing:
- Comprehensive testing has been performed to ensure the robustness and accuracy of email summarization.
- Unit tests cover individual components such as email retrieval, summarization, and email sending functionalities.

## Screenshots:
Screenshot 2024-03-24 at 11 27 23 PM

Chrome Extension

Screenshot 2024-03-24 at 11 31 11 PM

TL;DR email

### CODE REVIEW

**Review by mrinoybanerjee — APPROVED** [review]
Repo structure looks amazing! Nothing short of impressed by the chrome extension feature you have added! Code looks clean, beautiful and well commented. Way to go Abhishek! 

PS: Lets include dependency versioning in the requirements.txt file
